CDCgov/NEDSS-Modernization
# CNFT2-1155 Create Tab order_nbr

## Description

1. Fixes newly created Tab's order number.
2. Fixes Tab questionIdentifier
3. Splits the Tab management components. 

## Tickets

* [CNFT2-1155](https://cdc-nbs.atlassian.net/browse/CNFT2-1155)

## Steps to verify

1. Find a page in the database that you wish to add a new tab to. 
```sql
SELECT
	*
FROM
	wa_template
WHERE
	template_type = 'Draft';
```
2. Using the [modernization-api swagger ](http://localhost:9080/swagger-ui/index.html#/user-controller/loginUsingPOST). Login using the `superuser` account and copy the `token` from the response.
3. Using the [question-bank swagger](http://localhost:8095/nbs/page-builder/swagger-ui/#/tab-controller/createTabUsingPOST), send a request to create a new tab 
![image](https://github.com/CDCgov/NEDSS-Modernization/assets/109251240/c7b0cfbc-30e8-4593-b7ec-e3f5851b3422)
4. Run the following query to verify the tab is created. 
```sql
SELECT
	wa_ui_metadata_uid,
	nbs_ui_component_uid,
	question_label,
	order_nbr,
	display_ind
FROM
	wa_template
	JOIN  WA_UI_metadata ON wa_template.wa_template_uid = WA_UI_metadata.wa_template_uid
WHERE	
	 wa_template.wa_template_uid = 1000377 
	 AND wa_template.template_type = 'Draft'
	 AND order_nbr > 0
ORDER BY
	order_nbr ASC;
```
You will want to verify the order number is as expected (1 higher than previous row), the `question_label` matches the specified name provided, and that the `display_ind` is `T` if you set visibility to true or `F` if false.
![image](https://github.com/CDCgov/NEDSS-Modernization/assets/109251240/5abd4560-3afd-4e3a-9b7f-f3e5f0d6dcdd)


[CNFT2-1155]: https://cdc-nbs.atlassian.net/browse/CNFT2-1155?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
